topic_classifier/classifier.py

```
@@ -0,0 +1,152 @@
+        }
+
+        response = requests.post(LLM_API_URL, json=payload)
+        response.raise_for_status()
+        if response.status_code == 200:
+            return response.json()
+        else:
+            raise HTTPException(
+                status_code=response.status_code, detail="Error querying LLM API"
+            )
+    except requests.exceptions.RequestException as e:
+        raise HTTPException(status_code=503, detail=f"LLM API unreachable: {str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)